cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,26 +38,37 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
-	if !cmdUtil.IsValidProfile(profileName) {
+	// Unset the OC CLI context if present for the active profile
+	if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
+		err := cmdUtil.RemoveCurrentContext()
+		if err != nil {
+			if glog.V(2) {
+				fmt.Println(fmt.Sprintf("%s", err.Error()))
+			}
+		}
+	}
+
+	if cmdUtil.IsValidProfile(profileName) {
 		err := profileActions.SetActiveProfile(profileName)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
-	} else {
-		err := cmdUtil.SetOcContext(profileName)
+		fmt.Printf("Profile '%s' set as active profile.\n", profileName)
+
+		err = cmdUtil.SetOcContext(profileName)
 		if err != nil {
 			if glog.V(2) {
 				fmt.Println(fmt.Sprintf("%s", err.Error()))
 			}
-			fmt.Println(fmt.Sprintf("oc cli context could not changed for '%s'. %s", profileName, unableToSetOcContextErrorMessage))
```

Comment:
IMO`fmt.Println(fmt.Sprintf("oc cli context could not changed for '%s'. %s", profileName))` should be part of the above `if glog.V(2) {` , So that user will come know about context is not set when running debug logs.

Reply:
> IMO `fmt.Println(fmt.Sprintf("oc cli context could not changed for '%s'. %s", profileName))` should be part of the above if glog.V(2)

Done.